### ISSUE TITLE
Syncing a sample does not set the correct collectionDate

### ISSUE THREAD

**Issue by apetkau** [phac-nml/irida#256]
## Describe the bug

If a remote machine has a sample with a **Collection Date** filled in, and this gets synced to a local machine, the local date will be one day behind the remote date.

| Time | Remote | Local |
|------|--------|-------|
| 1    | Sample A (Collection Date = Jan 10, 2019) | None |
| 2    | Sample A (Jan 10, 2019) | Sample A (**Jan 9, 2019**) |

## Steps to reproduce the problem

1. Create a sample and set the **Collection Date**.
2. Sync the project the sample was created in.
3. Check the date of the sample in the synced project. It will be one day behind the original date.

## Expected behaviour

I expected the dates to be identical.

**Comment by apetkau** [comment]
Printing out the dates after syncing, I get `Tue Jan 15 18:00:00 CST 2019`.  This is (-600) from Jan 16 at midnight and our timezone offset is -0600, so I'm guessing this is a timezone interpretation issue.

**Comment by apetkau** [comment]
Some info at https://stackoverflow.com/questions/44542923/getting-wrong-time-in-date-while-using-spring-rest-service

**Comment by apetkau** [comment]
The `collectionDate` in transfered by the REST API is only at the level of date, so timezone issues can't be present here:

```
GET http://localhost:8080/api/projects/9/samples
{
  "resource" : {
    "resources" : [ {
      "sampleName" : "SampleA",
      "collectionDate" : "2019-01-15",
```

**Comment by apetkau** [comment]
I suspect it's difference between `java.util.date` and `java.sql.date`. On serializing to JSON it's using `java.sql.date`. But, on deserializing it is converting to `java.util.date`.

**Comment by tom114** [comment]
Fixed in #263
